src/test/kotlin/com/atlassian/performance/tools/hardware/tuning/HeapTuning.kt

```diff
@@ -0,0 +1,73 @@
+package com.atlassian.performance.tools.hardware.tuning
+
+import com.amazonaws.services.ec2.model.InstanceType.*
+import com.atlassian.performance.tools.hardware.ApplicationScale
+import com.atlassian.performance.tools.hardware.Hardware
+import com.atlassian.performance.tools.infrastructure.api.jira.JiraJvmArgs
+import com.atlassian.performance.tools.infrastructure.api.jira.JiraLaunchTimeouts
+import com.atlassian.performance.tools.infrastructure.api.jira.JiraNodeConfig
+import com.atlassian.performance.tools.infrastructure.api.jvm.JvmArg
+import java.time.Duration
+
+class HeapTuning : JiraNodeTuning {
+
+    override fun tune(
+        nodeConfig: JiraNodeConfig,
+        hardware: Hardware,
+        scale: ApplicationScale
+    ): JiraNodeConfig {
+        return if (scale.description == "Jira XL profile") {
+            JiraNodeConfig.Builder(nodeConfig)
+                .jvmArgs(
+                    bumpHeap(50, hardware)
+                )
+                .launchTimeouts(
+                    JiraLaunchTimeouts.Builder()
+                        .initTimeout(Duration.ofMinutes(7))
+                        .offlineTimeout(Duration.ofMinutes(15))
+                        .build()
+                )
+                .build()
+        } else {
+            nodeConfig
+        }
+    }
+
+    private fun bumpHeap(
+        desiredGigabytes: Int,
+        hardware: Hardware
+    ): JiraJvmArgs {
+        val maxGigabytes = when (hardware.jira) {
+            C52xlarge -> 12
+            C54xlarge -> 25
+            C48xlarge -> 50
+            C59xlarge -> 60
```

Comment:
I like that we have different heap per instance type. I think anything over 50G might not be necessary. We have not seen much GC problem with 50G according to Jprofiler result.

Reply:
Note that this is `maxGigabytes`.
We actually `coerceAtMost(desiredGigabytes)`, so we'll use 50 GB even though more is possible.